cmake/checks/check_02_compiler_features.cmake

```diff
@@ -387,7 +397,9 @@ if(NOT CMAKE_CXX_COMPILER_ID MATCHES "MSVC")
     "
     DEAL_II_COMPILER_HAS_FUSE_LD_GOLD)
 
-  if(DEAL_II_COMPILER_HAS_FUSE_LD_LLD)
+  if(DEAL_II_COMPILER_HAS_FUSE_LD_MOLD)
+    add_flags(DEAL_II_LINKER_FLAGS "-fuse-ld=mold")
+  elseif(DEAL_II_COMPILER_HAS_FUSE_LD_LLD)
     add_flags(DEAL_II_LINKER_FLAGS "-fuse-ld=lld")
   elseif(DEAL_II_COMPILER_HAS_FUSE_LD_GOLD)
```

Comment:
You're placing mold above lld. We know from experiments that gold is faster than ld, and I suspect someone has already tested that lld is faster than gold. But have you actually tested that mold is faster than lld.

mold being new also makes me worried that it has already implemented all of the arcane stuff that has been put on the shoulders of linkers over the years. Is it reliable?

Reply:
I share the concern about mold being prime time ready.

In fact, even for lld we end up dropping `-fuse-ld=lld` again for a considerable number of configurations. All at the expense of a significant number of CMake script and configuration time.

I am currently contemplating about all of this and I am currently leaning towards removing a significant amount of our "automagic" configuration behavior. For example, it would be simply be enough to test for `ld.gold` as a substitute for `ld.bfd` but leave the rest for the user to decide.

After all, simply adding `-fuse-ld=mold` to the CMake variable `DEAL_II_LINKER_FLAGS` or the environment variable `LDFLAGS` achieves the same thing.